nemo/collections/nlp/models/language_modeling/megatron_finetune_model.py

```diff
@@ -333,58 +306,73 @@ def fwd_bwd_step(self, dataloader_iter, batch_idx, forward_only):
         return loss_mean
 
     def inference_step(self, dataloader_iter, batch_idx: int, mode: str, dataloader_idx=0):
-        # Regular finetuning datasets will return a list of dicts for each microbatch.
-        # But T0 datasets will return a single dict for the global batch.
-        batch = next(dataloader_iter)
-        batch_has_lang_information = isinstance(batch, list) and len(batch[0]) == 7
-        data_cfg = self.cfg.data.validation_ds if mode == 'validation' else self.cfg.data.test_ds
-
-        self._reconfigure_and_process_inference_batch(batch, data_cfg)
-
-        # NOTE: There could be extra keys in the processed_batch dictionary such as "langs" for XNLI,
-        # this will be ignored.
-        loss = self.fwd_bwd_step(itertools.chain([batch]), batch_idx, forward_only=True)
-
-        predicted_token_ids, _ = self.decode(
-            tokens_enc=batch['text_enc'],
-            enc_mask=batch['enc_mask'],
-            num_tokens_to_generate=30,
-            bos_id=self.tokenizer.pad_id if data_cfg.get('replace_bos_with_pad', False) else self.tokenizer.bos_id,
-        )
+        # Add try except since dataloader_iter in PTL 2.0 doesnt catch the end of the iterator
```

Comment:
ugh... this try catch looks ugly...
I dont understand the rationale why PTL removed the StopIteration check
Instead of doing this on every step function, can we handle this try-catch at a level above? maybe directly handle this in base class or create a function that calls our step functions that puts everything in try except?

Reply:
Deferred to subsequent PR.